File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        unordered_map<int, deque<int>> idx;
        int n = temperatures.size();
        vector<int> ans(n);
        for(int i=0;i<n;++i){
            idx[temperatures[i]].push_back(i);
        }
        for(int i=0;i<n;++i){
            ans[i] = 1e7;
            idx[temperatures[i]].pop_front();
            for(int temp=temperatures[i]+1;temp<101;++temp){
                if(!idx[temp].empty()){
                    ans[i] = min(ans[i], idx[temp][0] - i);
                }
            }
            if(ans[i] == 1e7){
                ans[i] = 0;
            }
        }
        return ans;
    }
};

File: 0739-daily-temperatures/0739-daily-temperatures.cs
public class Solution {
    public int[] DailyTemperatures(int[] temperatures) {
        int n = temperatures.Length;
        Dictionary<int, Queue<int>> d = new Dictionary<int, Queue<int>>();
        int[] ans = new int[n];
        for(int i=0;i<n;i++)
        {
            ans[i] = 1000000;
            int x = temperatures[i];
            if(!d.ContainsKey(x))
            {
                d.Add(x, new Queue<int>());
            }
            d[x].Enqueue(i);
        }
        for(int i=0;i<n;i++)
        {
            int x = temperatures[i];
            d[x].Dequeue();
            for(int temp=x+1;temp<=100;temp++)
            {
                if(d.ContainsKey(temp) && d[temp].Count()!=0)
                {
                    ans[i] = Math.Min(ans[i], d[temp].Peek() - i);
                }
            }
            if(ans[i] == 1000000)
            {
                ans[i] = 0;
            }
        }
        return ans;
    }
}